Service/DataService.php

```diff
@@ -2010,4 +2023,101 @@ public function createAndMapIndex(Environment $environment): void
             'name' => $environment->getAlias()
         ]);
     }
+
+    public function getDataLinks(string $contentTypesCommaList, array $businessIds): array
+    {
+        $items = [];
+        $ouuids = [];
+        foreach ($businessIds as $businessId) {
+            if (isset($this->cacheOuuids[$contentTypesCommaList][$businessId])) {
+                $ouuids[$businessId] = $this->cacheOuuids[$contentTypesCommaList][$businessId];
+            } else {
+                $items[] = $businessId;
+                $ouuids[$businessId] = $businessId;
+            }
+        }
+
+        foreach ($this->contentTypeService->getByNames(explode(',', $contentTypesCommaList)) as $contentType) {
+            if ($contentType->getBusinessIdField() && count($ouuids) > 0) {
+                $result = $this->client->search([
```

Comment:
```suggestion
        foreach (explode(',', $contentTypesCommaList) as $contentTypeName) {
            $contentType = $this->contentTypeService->getByName($contentTypeName);
            if ($contentType->getBusinessIdField() && count($ouuids) > 0) {
                $result = $this->client->search([
```

The getByNames function is not needed. Adding10 lines of code that are fixed here with 1 extra line of code

Reply:
9e53b2e7dded6dba8a74028e301873ec791a5c71